### PR TITLE
cli: implement support for credential and notification association

### PR DESCRIPTION
**Pull request by ryanpetrello**

### CODE REVIEW

**Comment by ryanpetrello** [comment]
Writing some tests for this now...

File: awxkit/awxkit/cli/custom.py [hunk 1]
```diff
@@ -182,6 +183,111 @@ class AdhocCommandStdout(HasStdout, CustomAction):
...
+            endpoint, _ = self.targets[k]
+            try:
+                self.page.get().related[endpoint].post(
+                    {'id': v, self.action: True}
+                )
+            except NoContent:
+                # we expect to enter this block because these endpoints return
+                # HTTP 204 on success
+                pass
+            return self.page.get().related[endpoint].get()
```
Comment (jakemcdermott) [inline awxkit/awxkit/cli/custom.py:211]: Should this `return` be in the loop or after it?

Comment (ryanpetrello) [reply awxkit/awxkit/cli/custom.py:211]: Inside it - the flags to this command, which represent "which object am I attaching/detaching", are mutually exclusive:

https://github.com/ansible/awx/pull/4624/files/88dfcaa439d9cdac4a9c5de0a09771f026cf1fa6#diff-a8a950fa8ea340c107209be099b3710aR192

...so the goal is to find the first (and only) match in the list and return the results of the attach/detach of that item.

**Review by jakemcdermott — APPROVED** [review]

**Review by kdelee — APPROVED** [review]
:ship:

**Pull request merged**